src/decode/stream.rs

```diff
@@ -155,24 +156,23 @@ where
     ) -> crate::error::Result<State<W>> {
         match LzmaParams::read_header(&mut input, options) {
             Ok(params) => {
-                let decoder = if let Some(memlimit) = options.memlimit {
-                    new_circular_with_memlimit(output, params, memlimit)
-                } else {
-                    new_circular(output, params)
-                }?;
-
+                let dict_size = params.dict_size;
+                let decoder = new_circular(params)?;
+                let output =LzCircularBuffer::from_stream_with_memlimit(output, dict_size as usize,
```

Comment:
Is `LzCircularBuffer::from_stream` still used after this change? If not, let's just remove it and rename `from_stream_with_memlimit` to `from_stream`.

Reply:
Renamed `LzCircularBuffer::from_stream_with_memlimit ` to `from_stream`. Also renamed `LzAccumBuffer::from_stream_with_memlimit` to `from_stream` to mirror the `LzCircularBuffer` API. The only consumer of `LzAccumBuffer::from_stream` was `lzma2`, where I just passed `usize::MAX` as the default limit.